### PR TITLE
Updates Objective-C proto plugin

### PR DESCRIPTION
**Pull request by jcanizales**
It now generates a surface according to the last design (except for some nits), an implementation compatible with our current version of the proto-gRPC runtime, and uses proto3.

Known missing things:
- The input and output messages for rpc methods are referred-to using the prefix of the rpc service. Because one such message can be defined in another file, we have to use the prefix specified in that file instead.
- We're not propagating the comments from the .proto file.
- We can't yet generate code for services defined in a 2.6 file, nor for rpc methods whose input or output types are defined in a 2.6 file.
- We're naming "handler" the block parameter for both single-response methods and stream-response methods. In the surface design we agreed to call the latter "eventHandler".

### CODE REVIEW

**Comment by jcanizales** [comment]
(I recommend split view in the diff page, because it's hard to follow with the inline diff view).

**Comment by jcanizales** [comment]
Ah, also, I'm stumbling upon what I think is a bug in the protoc libraries linked in the plugin, which wouldn't let me append code to existing files, nor insert code at insertion points of existing files. So as I can only write to new files, I worked around the filename-collision worry by using .pbrpc.h/m extension instead of .pbobjc.h/m.

**Comment by murgatroid99** [comment]
So, is the `Test.pb.m` file now generated with this plugin?

**Comment by murgatroid99** [comment]
The plugin is failing to build on Travis.

**Comment by jcanizales** [comment]
Oooooh "error: ‘const class google::protobuf::FileOptions’ has no member named ‘objc_class_prefix’" that means Travis isn't installing (the latest?) protoc v3 libraries.

**Comment by jcanizales** [comment]
`Test.pb.m`: Sorry, that change snuck in. I reverted my generated files for that code because it generates proto3 surface now, and it would conflict with the integration tests you're writing.

**Comment by nicolasnoble** [comment]
Re: versions of protobuf, see #1455.

**Comment by jcanizales** [comment]
Travis passes now :)
